cmake/InstallTPLs.cmake

```diff
@@ -5,88 +5,102 @@
+install(TARGETS PolyClipperAPI
+  EXPORT spheral_cxx-targets
+  DESTINATION lib/cmake)
+set_target_properties(PolyClipperAPI PROPERTIES EXPORT_NAME spheral::PolyClipperAPI)
 
 #-----------------------------------------------------------------------------------
 # Find pre-compiled TPLs
 #-----------------------------------------------------------------------------------
 
+# Use find_package to get axom (which brings in fmt) and patch fmt
+list(APPEND SPHERAL_BLT_DEPENDS axom)
+find_package(axom REQUIRED QUIET NO_DEFAULT_PATH PATHS ${axom_DIR}/lib/cmake)
+# Add fmt library to external library list
+list(APPEND SPHERAL_BLT_DEPENDS fmt)
+blt_patch_target(NAME fmt TREAT_INCLUDES_AS_SYSTEM ON)
+
 # Initialize TPL options
 include(${SPHERAL_ROOT_DIR}/cmake/spheral/SpheralHandleTPL.cmake)
 
-# These libs are always needed
-Spheral_Handle_TPL(zlib spheral_depends cxx)
-Spheral_Handle_TPL(boost spheral_depends cxx)
-Spheral_Handle_TPL(eigen spheral_depends cxx)
-Spheral_Handle_TPL(qhull spheral_depends cxx)
-Spheral_Handle_TPL(silo spheral_depends cxx)
+# TPLs that must be imported
+list(APPEND SPHERAL_EXTERN_LIBS zlib boost eigen qhull silo hdf5 polytope)
 
-# AXOM PUlls in HDF5 and Conduit for us
-#Spheral_Handle_TPL(conduit spheral_depends cxx)
-find_package(axom REQUIRED QUIET NO_DEFAULT_PATH PATHS ${axom_DIR}/lib/cmake)
-if(axom_FOUND)
-  list(APPEND spheral_blt_cxx_depends axom fmt)
-  blt_patch_target(NAME fmt TREAT_INCLUDES_AS_SYSTEM On)
-  message(STATUS "Found axom: ${axom_DIR} (found version ${axom_VERSION})")
+if(ENABLE_ANEOS)
+  list(APPEND SPHERAL_EXTERN_LIBS aneos)
 endif()
-
-# Axom imports hdf5 lib but Spheral also requires hdf5_hl
-Spheral_Handle_TPL(hdf5 spheral_depends cxx)
-
-# Some libraries are optional
-if (ENABLE_ANEOS)
-  Spheral_Handle_TPL(aneos spheral_depends cxx)
-endif()
-if (ENABLE_OPENSUBDIV)
-  Spheral_Handle_TPL(opensubdiv spheral_depends cxx)
+if(ENABLE_OPENSUBDIV)
+  list(APPEND SPHERAL_EXTERN_LIBS opensubdiv)
 endif()
 if(ENABLE_TIMER)
-  Spheral_Handle_TPL(caliper spheral_depends cxx)
+  list(APPEND SPHERAL_EXTERN_LIBS caliper)
 endif()
 
-# Only needed when building the python interface of spheral
-if(NOT ENABLE_CXXONLY)
-  Spheral_Handle_TPL(python spheral_depends cxx)
-  #Spheral_Handle_TPL(pyb11generator spheral_depends py)
-  #Spheral_Handle_TPL(pybind11 spheral_depends py)
-endif()
-
-Spheral_Handle_TPL(polytope spheral_depends cxx)
-#Spheral_Handle_TPL(polyclipper spheral_depends cxx)
+# Create target library for each external library
+foreach(lib ${SPHERAL_EXTERN_LIBS})
+  Spheral_Handle_TPL(${lib} ${TPL_SPHERAL_CMAKE_DIR})
+  list(APPEND SPHERAL_BLT_DEPENDS ${lib})
+endforeach()
+
+# Install each TPL target library
+foreach(lib ${SPHERAL_EXTERN_LIBS})
+  get_target_property(_is_imported ${lib} IMPORTED)
+  if(NOT ${_is_imported})
+    install(TARGETS ${lib}
+      EXPORT spheral_cxx-targets
+      DESTINATION lib/cmake)
+    set_target_properties(${lib} PROPERTIES EXPORT_NAME spheral::${lib})
+  endif()
```

Comment:
Could this be moved into the Spheral_Handle_TPL function. It would remove some code duplication